packages/react-charting/src/components/AreaChart/AreaChart.base.tsx

```diff
@@ -419,42 +418,28 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
     this._chart = this._drawGraph(containerHeight, xAxis, yAxis, xElement!);
   };
 
-  private _onLegendClick(customMessage: string): void {
-    if (this.state.isLegendSelected) {
-      if (this.state.activeLegend === customMessage) {
-        this.setState({
-          isLegendSelected: false,
-          activeLegend: '',
-        });
-      } else {
-        this.setState({
-          activeLegend: customMessage,
-        });
-      }
+  private _onLegendClick(legend: string): void {
+    if (this.state.selectedLegend === legend) {
+      this.setState({
+        selectedLegend: '',
+      });
     } else {
       this.setState({
-        activeLegend: customMessage,
+        selectedLegend: legend,
       });
     }
   }
 
-  private _onLegendHover(customMessage: string): void {
-    if (this.state.isLegendSelected === false) {
-      this.setState({
-        activeLegend: customMessage,
-        isLegendHovered: true,
-      });
-    }
+  private _onLegendHover(legend: string): void {
+    this.setState({
+      activeLegend: legend,
+    });
   }
 
-  private _onLegendLeave(isLegendFocused?: boolean): void {
```

Comment:
Do you need to keep track of isLegendFocused flag.

Reply:
No, it is not needed in the new design